runtime/oti/j9nonbuilder.h

```diff
@@ -2054,12 +2054,7 @@ typedef struct J9BCTranslationData {
 #define BCT_Java7MajorVersionShifted  0x33000000
 #define BCT_Java8MajorVersionShifted  0x34000000
 #define BCT_Java9MajorVersionShifted  0x35000000
-#define BCT_Java10MajorVersionShifted 0x36000000
 #define BCT_Java11MajorVersionShifted 0x37000000
-#define BCT_Java12MajorVersionShifted 0x38000000
-#define BCT_Java13MajorVersionShifted 0x39000000
-#define BCT_Java14MajorVersionShifted 0x3A000000
-#define BCT_Java15MajorVersionShifted 0x3B000000
```

Comment:
I propose we remove all `BCT_JavaXXMajorVersionShifted` macros and replace them with this:
```
/* A given Java feature version uses class-file format (version) + 44. */
#define BCT_JavaMajorVersionShifted(java_version) ((44 + (I_32)(java_version)) << BCT_MajorClassFileVersionMaskShift)
/* When adding support for a new Java feature version, update this appropriately. */
#define BCT_JavaMaxMajorVersionShifted BCT_JavaMajorVersionShifted(18)
```
Uses of `BCT_JavaXXMajorVersionShifted` become `BCT_JavaMajorVersionShifted(XX)` where `XX` is a numeric literal like `8` or `11`.

The constant 44 also appears in openjdk: https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/make/autoconf/version-numbers#L37

Reply:
I am slightly concerned about the extra runtime cost introduced by this macro. 
It doesn't seem fit this PR just aiming for cleanup.
Does it make sense?

Reply:
There should be exactly zero runtime cost to this. Compilers will fold the expression into exactly the same value.

Reply:
Agreed, updated.